atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/cleaner/ScrubberTest.java

```diff
@@ -143,6 +143,41 @@ public void testScrubQueueIsCleared() {
         assertThat(scrubQueue).isEmpty();
     }
 
+    @Test
+    public void scrubberIsResilientToTableDeletion() {
+        Cell cell1 = Cell.create(new byte[] {1}, new byte[] {2});
+        Cell cell2 = Cell.create(new byte[] {2}, new byte[] {3});
+        Cell cell3 = Cell.create(new byte[] {3}, new byte[] {4});
+        TableReference tableRef = TableReference.createFromFullyQualifiedName("foo.bar");
+        kvs.createTable(tableRef, new byte[] {});
+        kvs.putWithTimestamps(
+                tableRef,
+                ImmutableMultimap.<Cell, Value>builder()
+                        .put(cell1, Value.create(new byte[] {3}, 10))
+                        .put(cell1, Value.create(new byte[] {4}, 20))
+                        .put(cell2, Value.create(new byte[] {4}, 30))
+                        .put(cell2, Value.create(new byte[] {5}, 40))
+                        .put(cell2, Value.create(new byte[] {6}, 50))
+                        .put(cell3, Value.create(new byte[] {7}, 60))
+                        .build());
+        transactions.putUnlessExists(10, 15);
+        transactions.putUnlessExists(20, 25);
+        transactions.putUnlessExists(30, 35);
+        transactions.putUnlessExists(50, 55);
+        transactions.putUnlessExists(60, 65);
+        scrubStore.queueCellsForScrubbing(ImmutableMultimap.of(cell1, tableRef), 20, 100);
+        scrubStore.queueCellsForScrubbing(ImmutableMultimap.of(cell2, tableRef), 50, 100);
+        scrubStore.queueCellsForScrubbing(ImmutableMultimap.of(cell3, tableRef), 60, 100);
```

Comment:
Note that this is a shorter list than the cells queued in the other test - overwritten cells don't appear to be processed properly by a single iteration of the scrubber. I'm fairly confident that these will be dealt with by future scrubber iterations.